activitysim/abm/models/vehicle_type_choice.py

```diff
@@ -0,0 +1,444 @@
+
+    Returns
+    -------
+    alts_wide : pd.DataFrame in wide format expanded using pandas get_dummies function
+    alts_long : pd.DataFrame in long format
+    """
+    cat_cols = list(alts_cats_dict.keys())  # e.g. fuel type, body type, age
+    alts_long = pd.DataFrame(
+        list(itertools.product(*alts_cats_dict.values())),
+        columns=alts_cats_dict.keys()).astype(str)
+    alts_wide = pd.get_dummies(alts_long)  # rows will sum to num_cats
+
+    # store alts in primary configs dir -- is this the best place for it?
+    configs_dirs = inject.get_injectable("configs_dir")
+    configs_dirs = [configs_dirs] if isinstance(configs_dirs, str) else configs_dirs
+
+    alts_wide = pd.concat([alts_wide, alts_long], axis=1)
+    return alts_wide, alts_long
+
+
+def append_probabilistic_vehtype_type_choices(
+        choices, probs_spec_file, trace_label):
+    """
+    Selects a fuel type for the provided body type and age of the vehicle.
+    Makes probabilitstic choices based on the `PROBS_SPEC` file.
+
+    Parameters
+    ----------
+    choices : pandas.DataFrame
+        selection of {body_type}_{age} to append vehicle type to
+    probs_spec_file : str
+    trace_label : str
+
+    Returns
+    -------
+    choices : pandas.DataFrame
+        table of chosen vehicle types
+    """
+    # name of first column must be "vehicle_type"
+    probs_spec = pd.read_csv(
+        config.config_file_path(probs_spec_file), comment='#')
+
+    # left join vehicles to probs
+    choosers = pd.merge(
+        choices.reset_index(), probs_spec,
+        on='vehicle_type',
+        how='left',
+        indicator=True).set_index('vehicle_id')
+
+    # checking to make sure all alternatives have probabilities
+    missing_alts = choosers.loc[choosers._merge == 'left_only', ['vehicle_type']]
+    assert len(missing_alts) == 0, \
+        f"missing probabilities for alternatives:\n {missing_alts}"
+    choosers.drop(columns='_merge', inplace=True)
+    del choosers['vehicle_type']
+
+    # probs should sum to 1 with residual probs resulting in choice of 'fail'
+    chooser_probs = choosers.div(choosers.sum(axis=1), axis=0).fillna(0)
+    chooser_probs['fail'] = 1 - chooser_probs.sum(axis=1).clip(0, 1)
+
+    # make probabilistic choices
+    prob_choices, rands = logit.make_choices(chooser_probs, trace_label=trace_label, trace_choosers=choosers)
+
+    # convert alt choice index to vehicle type attribute
+    prob_choices = chooser_probs.columns[prob_choices.values].to_series(index=prob_choices.index)
+    failed = (prob_choices == chooser_probs.columns.get_loc('fail'))
+    prob_choices = prob_choices.where(~failed, "NOT CHOSEN")
+
+    # add new attribute to logit choice vehicle types
+    choices['vehicle_type'] = choices['vehicle_type'] + '_' + prob_choices
+
+    return choices
+
+
+def annotate_vehicle_type_choice(model_settings, trace_label):
+    """
+    Contains the calls to optionally annotate the persons, households, or
```

Comment:
Its more legible to call this function to annotate iff the config says to.

Even more legible to separate out methods for annotating persons vs hhs. See: https://en.wikipedia.org/wiki/Single-responsibility_principle

Reply:
Split each annotate table into its own function that only gets called if stated in the config.